src/tbb/dynamic_link.cpp

```diff
@@ -387,7 +387,11 @@ namespace r1 {
     #endif /* !__TBB_DYNAMIC_LOAD_ENABLED */
         // RTLD_GLOBAL - to guarantee that old TBB will find the loaded library
         // RTLD_NOLOAD - not to load the library without the full path
-        library_handle = dlopen(library, RTLD_LAZY | RTLD_GLOBAL | RTLD_NOLOAD);
+        #ifndef __HAIKU__
+		    library_handle = dlopen(library, RTLD_LAZY | RTLD_GLOBAL | RTLD_NOLOAD);
+        #else
+		    library_handle = dlopen(library, RTLD_LAZY | RTLD_GLOBAL);
```

Comment:
Otherwise:
```
/RAMDisk/oneTBB-2021.1.1/src/tbb/dynamic_link.cpp:390:68: error: 'RTLD_NOLOAD' was not declared in this scope
         library_handle = dlopen(library, RTLD_LAZY | RTLD_GLOBAL | RTLD_NOLOAD);
                                                                    ^~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
```
Grepping the Haiku headers returns nothing, presumably not supported on Haiku.

Reply:
I am not sure that it is a correct change. It will lead to library loading from arbitrary path. In this place we want to query already available symbols without library load. Consider the following change
```suggestion
		    library_handle = dlopen(nullptr, RTLD_LAZY);
```